scipy/spatial/distance.py

```diff
@@ -1092,26 +1156,36 @@ def sokalsneath(u, v):
 _SIMPLE_PDIST = {}
 
 for wrap_name, names, typ in [
-    ("bray_curtis", ['braycurtis'], "double"),
-    ("canberra", ['canberra'], "double"),
-    ("chebyshev", ['chebychev', 'chebyshev', 'cheby', 'cheb', 'ch'], "double"),
-    ("city_block", ["cityblock", "cblock", "cb", "c"], "double"),
-    ("euclidean", ["euclidean", "euclid", "eu", "e"], "double"),
-    ("sqeuclidean", ["sqeuclidean", "sqe", "sqeuclid"], "double"),
-    ("dice", ["dice"], "bool"),
-    ("kulsinski", ["kulsinski"], "bool"),
-    ("rogerstanimoto", ["rogerstanimoto"], "bool"),
-    ("russellrao", ["russellrao"], "bool"),
-    ("sokalmichener", ["sokalmichener"], "bool"),
-    ("sokalsneath", ["sokalsneath"], "bool"),
-    ("yule", ["yule"], "bool"),
+    ('bray_curtis', ['braycurtis'], 'double'),
+    ('canberra', ['canberra'], 'double'),
+    ('chebyshev', ['chebychev', 'chebyshev', 'cheby', 'cheb', 'ch'], 'double'),
+    ('city_block', ['cityblock', 'cblock', 'cb', 'c'], 'double'),
+    ('correlation', ['correlation', 'co'], 'double'),
+    ('cosine', ['cosine', 'cos'], 'double'),
+    ('dice', ['dice'], 'bool'),
+    ('euclidean', ['euclidean', 'euclid', 'eu', 'e'], 'double'),
+    ('kulsinski', ['kulsinski'], 'bool'),
+    ('mahalanobis', ['mahalanobis', 'mahal', 'mah'], 'double'),
+    ('minkowski',['minkowski', 'mi', 'm'], 'double'),
+    ('rogerstanimoto', ['rogerstanimoto'], 'bool'),
+    ('russellrao', ['russellrao'], 'bool'),
+    ('seuclidean', ['seuclidean', 'se', 's'], 'double'),
+    ('sokalmichener', ['sokalmichener'], 'bool'),
+    ('sokalsneath', ['sokalsneath'], 'bool'),
+    ('sqeuclidean', ['sqeuclidean', 'sqe', 'sqeuclid'], 'double'),
+    ('weighted_minkowski', ['wminkowski', 'wmi', 'wm', 'wpnorm'], 'double'),
+    ('yule', ['yule'], 'bool'),
 ]:
     converter = {"bool": _convert_to_bool,
                  "double": _convert_to_double}[typ]
-    fn_name = {"bool": "%s_bool_wrap",
-               "double": "%s_wrap"}[typ] % wrap_name
-    cdist_fn = getattr(_distance_wrap, "cdist_%s" % fn_name)
-    pdist_fn = getattr(_distance_wrap, "pdist_%s" % fn_name)
+    if wrap_name in ['cosine', 'correlation']:
+        cdist_fn = eval('_%s_cdist_wrap' % wrap_name)
+        pdist_fn = eval('_%s_pdist_wrap' % wrap_name)
```

Comment:
I don't like the `eval` here. Maybe you could look up the name in `globals()` (or a more appropriate namespace).